lib/connect/index.js

```diff
@@ -68,7 +67,7 @@ function connect (brokerUrl, opts) {
       parsed.port = Number(parsed.port)
     }
 
-    opts = xtend(parsed, opts)
```

Comment:
Good call to get rid of dependencies that are no longer needed due to newer JS features.